context/AuthContext.js

```diff
@@ -1,19 +1,49 @@
-import React, { children, createContext, useState } from "react";
+import React, { createContext, useState, useEffect } from "react";
+import AsyncStorage from "@react-native-async-storage/async-storage";
 export const AuthContext = createContext();
 export const AuthProvider = ({ children }) => {
-  const [isLoding, setIsLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(true);
   const [userToken, setUserToken] = useState(null);
-  const authFinish = (token) => {
-    setUserToken(token);
+
+  const authFinish = (userUid) => {
+    setUserToken(userUid);
     setIsLoading(false);
-    console.log(token);
+    storeUserToken(userUid);
   };
+
+  const retrieveData = async () => {
+    try {
+      const value = await AsyncStorage.getItem("userToken");
+      if (value !== null) {
+        return value;
+      }
+    } catch (error) {
+      console.error(error);
+    }
+    return null;
+  };
+
+  useEffect(() => {
+    retrieveData().then((token) => {
+      setUserToken(token);
+    });
+  }, []);
+
   const logout = () => {
```

Comment:
while logging out remove token from async storage as well or on next load the user would get logged in again